integration/webhook_test.go

```diff
@@ -0,0 +1,175 @@
+				Type: addonsv1alpha1.OLMAllNamespaces,
+			}),
+			err: newStatusError(".spec.install.allNamespaces is required " +
+				"when .spec.install.type = AllNamespaces"),
+		},
+		{
+			addon: newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OLMOwnNamespace,
+				OLMOwnNamespace: &addonsv1alpha1.AddonInstallOLMOwnNamespace{
+					AddonInstallOLMCommon: addonsv1alpha1.AddonInstallOLMCommon{
+						Namespace:          "reference-addon",
+						PackageName:        ADDON_NAME,
+						Channel:            "alpha",
+						CatalogSourceImage: CATALOG_SOURCE_URL,
+					},
+				},
+			}),
+			err: nil,
+		},
+		{
+			addon: newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OLMAllNamespaces,
+				OLMAllNamespaces: &addonsv1alpha1.AddonInstallOLMAllNamespaces{
+					AddonInstallOLMCommon: addonsv1alpha1.AddonInstallOLMCommon{
+						Namespace:          "reference-addon",
+						PackageName:        ADDON_NAME,
+						Channel:            "alpha",
+						CatalogSourceImage: CATALOG_SOURCE_URL,
+					},
+				},
+			}),
+			err: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := integration.Client.Create(ctx, tc.addon)
+
+		if tc.err != nil {
+			if !reflect.DeepEqual(err, tc.err) {
+				log.Fatalf("unexpected error: %v\nexpected: %v",
+					err, tc.err)
+			}
+		} else {
+			if err != nil {
+				log.Fatalf("expected nil error, got: %v", err)
+			}
+
+			// clean-up addon
+			err := integration.Client.Delete(ctx, tc.addon)
+			require.NoError(t, err)
+
+			err = integration.WaitToBeGone(t, 5*time.Minute, tc.addon)
+			require.NoError(t, err, "wait for Addon to be deleted")
+		}
+	}
+}
+
+func TestAddonSpecImmutability(t *testing.T) {
+	ctx := context.Background()
+
+	addon := newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+		Type: addonsv1alpha1.OLMOwnNamespace,
+		OLMOwnNamespace: &addonsv1alpha1.AddonInstallOLMOwnNamespace{
+			AddonInstallOLMCommon: addonsv1alpha1.AddonInstallOLMCommon{
+				Namespace:          "reference-addon",
+				PackageName:        ADDON_NAME,
+				Channel:            "alpha",
+				CatalogSourceImage: CATALOG_SOURCE_URL,
+			},
+		},
+	})
+
+	err := integration.Client.Create(ctx, addon)
+	require.NoError(t, err)
+
+	// try to update immutable spec
+
+	resourceVersion := addon.GetResourceVersion()
+	addon = newAddonWithInstallSpec(addonsv1alpha1.AddonInstallSpec{
+		Type: addonsv1alpha1.OLMOwnNamespace,
+		OLMOwnNamespace: &addonsv1alpha1.AddonInstallOLMOwnNamespace{
+			AddonInstallOLMCommon: addonsv1alpha1.AddonInstallOLMCommon{
+				Namespace:          "reference-addon",
+				PackageName:        ADDON_NAME,
+				Channel:            "beta", // changed
+				CatalogSourceImage: CATALOG_SOURCE_URL,
+			},
+		},
+	})
+	addon.SetResourceVersion(resourceVersion)
+
+	err = integration.Client.Update(ctx, addon)
+	expectedErr := newStatusError(".spec.install is an immutable field and cannot be updated")
+
+	if !reflect.DeepEqual(err, expectedErr) {
```

Comment:
this whole block can be exchanged with:
https://pkg.go.dev/github.com/stretchr/testify/require?utm_source=godoc#Equal

If not, there should be a doc string explaining why :)